### ISSUE TITLE
Serialization doesn't work when a document contains a reference

### ISSUE THREAD

**Issue by cjmyles** [wovalle/fireorm#95]
I know that there is a plan to add relationship support to Fireorm (https://github.com/wovalle/fireorm/issues/32), however I noticed that adding a reference to a document causes an error.

I copied the example from this [Medium article](https://medium.com/@wovalle/introducing-fireorm-ba7734644684) as follows which works fine:

```
@Collection()
class Band {
    id!: string;
    name!: string;
    formationYear!: number;
    genres!: Array<string>;
}

// Create a band
const band = new Band();
band.name = "A Perfect Circle";
band.formationYear = 1999;
band.genres = ['alternative-rock', 'hard-rock']

const bandDocument = await bandRepository.create(band);
console.log(`Created band with id: ${band.id}`);
```

As  soon as I go into the Firebase Console, locate this document and then add a new field which is a reference to another collection I get the following error:

```
TypeError: Cannot read property '_settings' of undefined
    at new Serializer (/node_modules/@google-cloud/firestore/build/src/serializer.js:46:23)
    at TransformOperationExecutor.transform 
```

Is this right, or is there any easy way to fix it? I have an existing Firestore database that contains references so I need Fireorm to fail gracefully with references until it's fully supported.

P.S I assume you are using [TypeORM](https://github.com/typeorm/typeorm/blob/master/docs/relations.md) as a reference for adding entity relations?

**Comment by wovalle** [comment]
Hey!

Yes, I actually found a similar bug when working on #82 which itself is contained in #58. The problem is that fireorm doesn't know how to instantiate the custom Reference class returned by firestore and it throws an error. The temporary/hack solution is to do something similar of [what I did to support geopoints](https://github.com/wovalle/fireorm/blob/master/src/BaseFirestoreRepository.spec.ts#L20-L21) (which is using the @Type Decorator to tell fireorm how to parse the class, that's explained [here](https://github.com/wovalle/fireorm#firebase-complex-data-types)). The long term solution is that fireorm know how to parse every firestore class and provide you a way to tell how to parse the things that it doesn't know, but that must wait until #58.

I have limited internet connectivity until the weekend, I'll check deeper then.

**Comment by cjmyles** [comment]
Hola! Thanks for the quick reply.

I'm happy with the temporary/hack solution, and seems like it should be easy enough to implement, however I'm still getting the same serialisation error with this code:

```
class Pillar { }

@Collection()
class Band {
    id!: string;
    name!: string;
    formationYear!: number;
    genres!: Array<string>;
    @Type(() => Pillar) pillar!: Pillar;
}
```

Where `pillar` is my Firebase reference to another collection. Any idea what I'm doing wrong?

**Comment by wovalle** [comment]
Hey! I was quite busy last week, I'll try to take a look at this in the coming days :)

**Comment by wovalle** [comment]
Hey!

I took a look at this issue and I'm afraid it'll a bit more cumbersome that I was expecting.

1st question: How do you add a "reference" field? I was looking in Firebase Console but "reference" is not inside the field types [see image](https://user-images.githubusercontent.com/7854116/67417452-1620f100-f5c9-11e9-84c8-6181ea6dfb1b.png).

2nd Q: Adding a reference via the console makes the query to return a [CollectionReference](https://firebase.google.com/docs/reference/js/firebase.firestore.CollectionReference) or a [Document Reference]()?

**Comment by cjmyles** [comment]
Hey, thanks for taking a look at this.

The "reference" type is actually underneath the "geopoint" entry in that dropdown - you just have to scroll down. Because the scrollbar isn't visible on chrome it isn't entirely obvious that it exists!

With regards to the reference, it's a DocumentReference type.

**Comment by wovalle** [comment]
Oh, didn't see the scrollbar. Will check it again, thanks!

**Comment by wovalle** [comment]
Hey! This was closed in #105. Now fireorm wont throw if it finds a DocumentReference! See [the integration test](https://github.com/wovalle/fireorm/blob/master/test/functional/5-document-references.spec.ts) to see how to use it!

PD. Don't forget to use @Type! (L11-12)

**Comment by cjmyles** [comment]
Amazing thanks dude! Quick work. Will check it out and let you know if I experience any issues.

**Comment by anhzf** [comment]
> Hey! This was closed in #105. Now fireorm wont throw if it finds a DocumentReference! See [the integration test](https://github.com/wovalle/fireorm/blob/master/test/functional/5-document-references.spec.ts) to see how to use it!
> 
> PD. Don't forget to use @type! (L11-12)

I'm sorry but I can't see it, please explain it in the documentation too
